src/Queue/Backend/Pdo/PdoQueueStoreAdapter.php

```diff
@@ -112,17 +112,26 @@ public function ack(MailJobInterface $mailJob)
             throw new InvalidCallException('PdoMailJob cannot be a new object to be acknowledged');
         }
 
-        $sqlText = 'UPDATE `%s`
+        $mailJob->incrementAttempt();
+        if($mailJob->getAttempt() > $_ENV['MAX_ATTEMPTS_DEFAULT']){
+            $sqlText = 'DELETE FROM mail_queue WHERE id = :max_attempt;';
+            $sql = sprintf($sqlText, $this->tableName);
+            $query = $this->getConnection()->getInstance()->prepare($sql);
+            $query->bindValue(':id', $mailJob->getId());
+        } else {
```

Comment:
avoid using `else ` when writing the algs, return query execute instead, them outside the if, the other case:

```
if($mailJob->getAttempt() > $_ENV['MAX_ATTEMPTS_DEFAULT']){
            $sqlText = 'DELETE FROM mail_queue WHERE id = :max_attempt;';
            $sql = sprintf($sqlText, $this->tableName);
            $query = $this->getConnection()->getInstance()->prepare($sql);
            $query->bindValue(':id', $mailJob->getId());

            return $query->execute();
        }

$sqlText = 'UPDATE `%s`
                SET `attempt`=:attempt, `state`=:state, `timeToSend`=:timeToSend, `sentTime`=:sentTime
                WHERE `id`=:id';

... rest of your code
```